repos/system_upgrade/el7toel8/actors/satelliteupgradefacts/actor.py

```diff
@@ -0,0 +1,133 @@
+                    additional_rpms.append(pkg.name.replace('tfm-rubygem-', 'rubygem-'))
+                elif pkg.name.startswith('tfm-pulpcore-python3-pulp'):
+                    additional_rpms.append(pkg.name.replace('tfm-pulpcore-python3-', 'python38-'))
+                elif pkg.name.startswith('foreman-installer') or pkg.name.startswith('satellite-installer'):
+                    additional_rpms.append(pkg.name)
+        self.produce(RpmTransactionTasks(to_install=additional_rpms))
+
+        if local_postgresql:
+            scl_psql_path = '/var/opt/rh/rh-postgresql12/lib/pgsql/data/'
+            scl_psql_stat = os.stat(scl_psql_path)
+            for nonscl_path in ['/var/lib/pgsql/data/', '/var/lib/pgsql/', '/var/lib/', '/']:
+                if os.path.exists(nonscl_path):
+                    nonscl_psql_stat = os.stat(nonscl_path)
+                    break
+
+            title = "Satellite PostgreSQL data migration"
+            flags = []
+            severity = reporting.Severity.MEDIUM
+
+            if scl_psql_stat.st_dev != nonscl_psql_stat.st_dev:
+                # get the current disk usage of the PostgreSQL data
+                scl_du_call = run(['du', '--block-size=1', '--summarize', scl_psql_path])
+                scl_used = int(scl_du_call['stdout'].split()[0])
+                # get the current free space on the target partition
+                nonscl_stat = os.statvfs(nonscl_path)
+                nonscl_free = nonscl_stat.f_bavail * nonscl_stat.f_frsize
+                if scl_used > nonscl_free:
+                    storage_message = "You currently don't have enough free storage to move the data. Automatic moving cannot be performed."
+                    flags = [reporting.Flags.INHIBITOR]
+                    severity = reporting.Severity.HIGH
+                else:
+                    storage_message = "You currently have enough free storage to move the data and this can be performed by the upgrade process."
+                summary = """
+                Your PostgreSQL data in {} is currently on a dedicated volume.
+                PostgreSQL on RHEL8 expects the data to live in /var/lib/pgsql/data.
+                {}
+                However, instead of moving the data over, you might want to consider manually adapting your mounts, so
+                that the contents of {} are available in /var/lib/pgsql/data.
+                """.format(scl_psql_path, storage_message, scl_psql_path)
+            else:
+                summary = "Your PostgreSQL data will be automatically migrated."
+
+            reporting.create_report([
+                reporting.Title(title),
+                reporting.Summary(textwrap.dedent(summary)),
+                reporting.Severity(severity),
+                reporting.Tags([]),
+                reporting.Flags(flags)
+            ])
```

Comment:
Reports need to be performed by a separate actor "Check" actor, however the trigger needs to be collected in here and you will have to create a custom model to send as a message. The so called check actor, is responsible for producing the report and may only use the information provided by the message. This is a convention with in leapp.